ra2ce/analysis/damages/damage_functions/damage_function_road_type_lane.py

```diff
@@ -18,56 +18,38 @@
     You should have received a copy of the GNU General Public License
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
-
-import logging
+from dataclasses import dataclass
 from pathlib import Path
 
 import pandas as pd
 
-from ra2ce.analysis.damages.damage.damage_fraction_uniform import DamageFractionUniform
-from ra2ce.analysis.damages.damage.max_damage import MaxDamageByRoadTypeByLane
+from ra2ce.analysis.damages.damage_functions.damage_fraction_uniform import (
+    DamageFractionUniform,
+)
+from ra2ce.analysis.damages.damage_functions.max_damage import MaxDamage
 
 
+@dataclass(kw_only=True)
 class DamageFunctionByRoadTypeByLane:
     """
     A damage function that has different max damages per road type, but a uniform damage_fraction curve
 
-
     The attributes need to be of the type:
-    self.max_damage (MaxDamage_byRoadType_byLane)
-    self.damage_fraction (DamageFractionHazardSeverityUniform)
-
+    self.max_damage (MaxDamage)
+    self.damage_fraction (DamageFractionUniform)
+    name (str)
     """
 
-    def __init__(
-        self,
-        max_damage: MaxDamageByRoadTypeByLane = None,
-        damage_fraction: DamageFractionUniform = None,
-        name: str = "",
-        hazard: str = "flood",
-        type: str = "depth_damage",
-        infra_type: str = "road",
-    ):
-        # Construct using the parent class __init__
-        self.name = name
-        self.hazard = hazard
-        self.type = type
-        self.infra_type = infra_type
-        self.max_damage = max_damage  # Should be a MaxDamage object
-        self.damage_fraction = (
-            damage_fraction  # Should be a DamageFractionHazardSeverity object
-        )
-        self.prefix = None  # Should be two characters long at maximum
-
-    def set_prefix(self):
-        self.prefix = self.name[0:2]
-        logging.info(
-            "The prefix: '{}' refers to curve name '{}' in the results".format(
-                self.prefix, self.name
-            )
-        )
+    max_damage: MaxDamage = None
+    damage_fraction: DamageFractionUniform = None
+    name: str = None
+
+    @property
+    def prefix(self) -> str:
+        return self.name[0:2]
 
-    def from_input_folder(self, folder_path: Path):
+    @classmethod
+    def from_input_folder(cls, name, folder_path: Path):
```

Comment:
missing typehint
```suggestion
    def from_input_folder(cls, name: str, folder_path: Path):
```

Reply:
When I look at the code here declared (dataclass fields / properties), and how it is being used (at least at the tests), I see a lack of consistency (and this might apply to the other refactored code). As an example, this `classmethod` is **requiring** a `name` whilst the dataclass says that its definition is entirely optional. Try to follow the following:
1. If it has default values:
    1. Create that object as you want (sometimes you can ignore the name) at the callers level with the arguments you want / need.
    2. If you need to modify its behavior, call the corresponding methods, at the callers level.
    3. You no longer need this class method.
2. Remove the default values. Then the classmethod is the only "shortcut" to create this object.

Reply:
2. seems most applicable